### ISSUE TITLE
cy.session does not restore chat session since upgrades from Cypress v10.10

### ISSUE THREAD

**Issue by jeanabanto** [cypress-io/cypress#25537]
### Current behavior

Up until Cypress v10.10, chat sessions are restored successfully with cy.session.

https://user-images.githubusercontent.com/2907631/213942300-3086a18d-cd1b-495c-9c37-f6c28b9b43d5.mov

Since Cypress 11, and the current v12, chat sessions are not restored successfully, and in effect starts a new chat session.

https://user-images.githubusercontent.com/2907631/213942337-c9ec2c24-56dc-4ff1-9dac-4bd70eab30c5.mov

### Desired behavior

Chat session should be restored successfully, i.e., chat history should not be deleted so that the app does not create a new session that in effect generates a new Zendesk ticket.

### Test code to reproduce

This repo currently has Cypress v10.10 installed, thus is expected to pass the test.
Upgrading Cypress to 11 or the latest release will fail the same test.

https://github.com/jeanabanto/cypress-zendesk-chat/blob/main/cypress/tests/chat.spec.ts

**Note:** If trial period in Zendesk expires and the account used in this repo stops working, you can easily create your own trial account in Zendesk with chat and replace the relevant environment variables.

### Cypress Version

12.3.0

### Node version

16.15.0

### Operating System

macOS 13.1 (22C65)

**Comment by ZachJW34** [comment]
Looked into this a bit yesterday. Like your video shows (and my testing), it looks like you're having issues with the `10.10.0` -> `10.11.0` upgrade. There were changes to both [cy.session](https://github.com/cypress-io/cypress/issues/24206) and [cy.origin](https://github.com/cypress-io/cypress/issues/23531) and you utilize both in your test (iframe widgets).

**Comment by jeanabanto** [comment]
Yes, this has been a blocker for our chat test scenarios. Would appreciate if this gets resolved soon or at least have any workaround for this (other than downgrading cypress).

**Comment by nikola-zitko** [comment]
I've just started using cy.session and I thought that I was doing something wrong. However, it looks like there is an issue inside Cypress itself. 
It would be nice to have this resolved soon as it's a blocker for our tests.

**Comment by bubafinder** [comment]
I noticed if I don't use cy.visit() in it(), works as expected, the session is restored successfully.
Before:
```
beforeEach(() => {
  cy.session([username, password], () => {
    cy.login(username, password);
  },
    {
      validate() { cy.visit('/'); }
    });
});

it('Test 1', () => {});

it('Test 1', () => {});
```

But if I add cy.visit() to the test, the session is not restored properly.
After:
```
beforeEach(() => {
  cy.session([username, password], () => {
    cy.login(username, password);
  },
    {
      validate() { cy.visit('/'); }
    });
});

it('Test 1', () => {
cy.visit('/');
});

it('Test 1', () => {
cy.visit('/');
});
```

**Comment by warrensplayer** [comment]
@jeanabanto I am picking this up from @ZachJW34  

Can you send an entire recording of your Cypress test in Cypress open mode that show the reporter as well.  When I try to reproduce this, this is afar as I can get with your code. (see video)


https://user-images.githubusercontent.com/1702361/216708482-087c0868-ecd4-4a12-b8ee-687223104b64.mov


@bubafinder Are you able to create a self-contained example project of what you are encountering?  Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.

**Comment by warrensplayer** [comment]
@jeanabanto Are there additional settings I need to change in my Zenhub account to get it to match how yours is set up?

**Comment by jeanabanto** [comment]
@warrensplayer I'm not sure why the chat is not loading properly for you so I've created a new trial environment. Please pull the updates from https://github.com/jeanabanto/cypress-zendesk-chat.

Also, here's a full recording of the test in v10.10

https://user-images.githubusercontent.com/2907631/216849215-9e80735d-4fe2-4cea-9616-ac897cf0920b.mp4

**Comment by marktnoonan** [comment]
Thanks for the reproduction @jeanabanto, I can see the test passing in 10.10 and failing in present version of Cypress, I have routed this to the team to look into the `cy.session()` issue.

**Comment by marktnoonan** [comment]
Note for future investigation - appears to start in 11.0.1, and is not affected by turning `testIsolation` off.

**Comment by bubafinder** [comment]
> I noticed if I don't use cy.visit() in it(), works as expected, the session is restored successfully. Before:
> 
> ```
> beforeEach(() => {
>   cy.session([username, password], () => {
>     cy.login(username, password);
>   },
>     {
>       validate() { cy.visit('/'); }
>     });
> });
> 
> it('Test 1', () => {});
> 
> it('Test 1', () => {});
> ```
> 
> But if I add cy.visit() to the test, the session is not restored properly. After:
> 
> ```
> beforeEach(() => {
>   cy.session([username, password], () => {
>     cy.login(username, password);
>   },
>     {
>       validate() { cy.visit('/'); }
>     });
> });
> 
> it('Test 1', () => {
> cy.visit('/');
> });
> 
> it('Test 1', () => {
> cy.visit('/');
> });
> ```

The quoted issue is probably caused by #25495
